### PR TITLE
Auto select random host port for a container

### LINKED ISSUE #4000: Unable to create container if broker process is up
It can be up for many reasons...
```
build> SECURITY WARNING: You are building a Docker image from 
Windows against a non-Windows Docker host. All files and directories added to build context 
will have '-rwxr-xr-x' permissions. It is recommended to double check and reset permissions 
for sensitive files and directories.
create> 23b672db28103d7a087f83cd024c9779e6a94da0cd8db60c65d603f569e19305
start 23b672db28103d7a087f83cd024c9779e6a94da0cd8db60c65d603f569e19305> 

ERROR: Error response from daemon: driver failed programming external connectivity on
 endpoint AA (ddad806e29e341c8d98dabf6f7700a05ec619fd6ed27535695214cdfdb13d9c9): 
Error starting userland proxy: Bind for 0.0.0.0:5444 failed: port is already allocated
Error: failed to start containers:
23b672db28103d7a087f83cd024c9779e6a94da0cd8db60c65d603f569e19305
```

### PR DESCRIPTION
**Pull request by karthiknadig**
Fixes #4000

### CODE REVIEW

File: src/Common/Core/Impl/Net/PortUtil.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+using System;
+using System.Net;
+using System.Net.Sockets;
+
+namespace Microsoft.Common.Core.Net {
+    public static class PortUtil {
+        private static Random _rand = new Random();
+        public static int GetAvailablePort(int minPort, int maxPort) {
+            while (true) {
+                int port = _rand.Next(minPort, maxPort + 1);
```
Comment (AlexanderSher) [inline src/Common/Core/Impl/Net/PortUtil.cs]: Can we do it for several ports in parallel?

File: src/Windows/R/Components/Impl/ContainerManager/Implementation/View/ContainerManagerControl.xaml [hunk 2]
```diff
@@ -152,7 +153,15 @@
                              ToolTip="{x:Static components:Resources.ContainerManager_CreateLocalDocker_VersionTooltip}"
                              AutomationProperties.Name="{x:Static components:Resources.ContainerManager_CreateLocalDocker_VersionTooltip}" />
 
-                    <WrapPanel Grid.Row="5" Grid.Column="1" Grid.ColumnSpan="2" Margin="0,0,0,0">
+                    <imaging:CrispImage Grid.Row="4" Grid.Column="0" Style="{StaticResource ValidityIndicatorStyle}"
+                                        cwpf:AttachedProperties.IsValid="{Binding Path=IsPortAvailable}" />
+
+                    <TextBox Grid.Row="4" Grid.Column="1" Style="{StaticResource TextBoxSingleLineStyle}"
+                             Text="{Binding Path=Port, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}"
+                             ToolTip="{x:Static components:Resources.ContainerManager_CreateLocalDocker_PortTooltip}"
```
Comment (AlexanderSher) [inline src/Windows/R/Components/Impl/ContainerManager/Implementation/View/ContainerManagerControl.xaml]: You may want to add `rcontrols:Watermark.TextBoxHint="Local port"` or something like that

File: src/R/Components/Impl/Containers/Implementation/ContainerManager.cs [hunk 3]
```diff
@@ -82,7 +82,7 @@ internal class ContainerManager : IContainerManager {
             }
         }
 
-        public async Task<IContainer> CreateLocalDockerAsync(string name, string username, string password, string version, CancellationToken cancellationToken = default(CancellationToken)) {
+        public async Task<IContainer> CreateLocalDockerAsync(string name, string username, string password, string version, int port, CancellationToken cancellationToken = default(CancellationToken)) {
             if (string.IsNullOrWhiteSpace(version)) {
                 version = "3.4.1";
```
Comment (MikhailArkhipov) [inline src/R/Components/Impl/Containers/Implementation/ContainerManager.cs]: Hm, why it is hardcoded? We can get latest from interpreter list perhaps? The list is local, but gives an idea what user is working with. Also, we could prepopulate with that number.

Comment (karthiknadig) [reply src/R/Components/Impl/Containers/Implementation/ContainerManager.cs]: It should be "latest". I will update this.

**Comment by MikhailArkhipov** [comment]
👍

**Pull request merged**